src/app/services/CalendarService.ts

```diff
@@ -0,0 +1,58 @@
+import {Injectable} from "@angular/core";
+import moment from "moment";
+import {CalendarItem} from "../models/CalendarItem";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CalendarService {
+  private readonly weekdaysShort = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
+
+  public createCalendarForYear(date: moment.Moment) {
+    return Array.from({ length: 12 }, (_, i) =>
+      this.createCalendar(date.clone().month(i))
+    );
+  }
+
+  public createCalendar(month: moment.Moment): Array<CalendarItem[]> {
+    const daysInMonth = month.daysInMonth();
+    const daysBefore = this.weekdaysShort.indexOf(month.startOf('months').format('ddd'));
+    const daysAfter = this.weekdaysShort.length - 1 - this.weekdaysShort
+      .indexOf(month.endOf('months').format('ddd'));
+
+    const calendar: CalendarItem[] = [];
+    const clone = month.startOf('months').clone().subtract(daysBefore, 'days');
+
+    for (let i = 0; i < daysBefore; i++) {
```

Comment:
`for of` mb?

Reply:
daysBefore, daysAfter, and daysInMonth are single numbers, not arrays. I didn't delete the for i loop; instead, I reduced the three loops into one using a ternary operator